GetCollisions Tapir command and GH component

tlorantfy opened the pull request:
https://github.com/user-attachments/assets/8089ee14-d8a2-4b27-b60b-e97404f8a905

tlorantfy commented:
@grzegorzwilk , could you please check the new GH node in the attached video?
Do you think this outputs will be OK?
So there are 2 ElementGuids output, the first item in the first output is collided with the first item in the second output and so on...

tlorantfy commented:
> I think the most common question I would use this component to answer is: Which indexes of the input list have an associated collision?
> 
> I am writing indexes because a lot of the time I’m not interested in filtering the actual GUID list that this component takes, but rather a different list — let’s say a list of polygon outlines, or elements from the official plugin.
> 
> The following formats preserve all information but make it easier to filter other lists based on collision results:
> 
> 1. A new branch for each input item, containing a list of booleans representing collisions.
> 2. A new branch for each input item, containing the indexes of the elements it collides with.
> 
> What do you think?

I'm still not convinced. It would be good to see a real workflow.
For me, using branches is very complicated, but I'm not an experienced GH user.

SzamosiMate commented:
> I'm still not convinced. It would be good to see a real workflow. For me, using branches is very complicated, but I'm not an experienced GH user.

I looked at the code, I misunderstood a bit how the componenet works from the video.

I think this is the part I'm on the fence with:

GSErrCode err = ACAPI_Element_GetCollisions (elemIds, elemIds, resultArray, collisionSettings);

I think we need 2 distinct input element groups. Most of the time you have distinct element sets when you do a collision detection (i.e. the classic structural frame x building services ducts), and if you don't, you can send the same set for the 2 parameters. 

After this, the Grasshopper component can function exactly like the "Clash" official componenet's First index/secound index outputs. With 2 sets there is no need for creating branches, as the user can do that by grafting one of the inputs getting exactly the output I outlined above.
<img width="1919" height="1079" alt="image" src="https://github.com/user-attachments/assets/4f307476-ae33-45c5-94ea-5b361f277ba1" />

tlorantfy commented:
Thanks @SzamosiMate for your thoughts!

I modified both the Tapir command and the GH component to be able to input 2 groups of elements (to detect collisions between these 2 groups).
And modified the GH component to return the indices also.

<img width="1902" height="1020" alt="image" src="https://github.com/user-attachments/assets/fb984622-969d-47b4-90a4-4c7c291e2441" />
